### PR TITLE
Switch from Cirrus CI to GitHub Actions

### PR DESCRIPTION
**Pull request by dominiklohmann**
After experiencing the delays of low availability for the macOS CI servers from Cirrus, we now plan to use GitHub Actions instead.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: recursive
+          ref: ${{ github.ref }}
+      - uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+      - name: Install Ubuntu dependencies
+        if: matrix.os == 'ubuntu-latest'
```
Comment (tobim) [inline .github/workflows/build.yml]: Is there a reason for switching away from debian?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          pip3 install --user cmake
+          ln -sf /usr/local/bin/cmake $(python3 -m site --user-site)/cmake/data/bin/cmake
+      - name: Install macOS dependencies
+        if: matrix.os == 'macos-latest'
+        run: |
+          brew install openssl cmake git ninja python libpcap jq tcpdump rsync pandoc apache-arrow
+      - name: Configure
+        run: |
+          echo "VAST-$(git describe)-$(uname -s)-${{ matrix.configure == '' && 'Release' || 'Debug' }}" > PACKAGE_NAME
+          ./configure --generator=Ninja --package-name=$(cat PACKAGE_NAME) --disable-arrow ${{ matrix.configure }}
```
Comment (tobim) [inline .github/workflows/build.yml]: Why aren't the existing CI scripts used here?

File: .github/workflows/build.yml (hunk 1 above)
Comment (0ortmann) [reply .github/workflows/build.yml]: There is no Debian in github actions.

File: .github/workflows/build.yml (hunk 2 above)
Comment (dominiklohmann) [reply .github/workflows/build.yml]: There are three reasons for this:
- The build matrix is easier to read if the configuration is either empty string or `--dev-mode`. 
- It is easier for a devops person without much knowledge about CMake (i.e., @0ortmann) not having to interact with `CMake -B...` manually.
- We recommend using `configure` in our README, so CI should use it, too.

File: .github/workflows/build.yml (hunk 1 above)
Comment (tobim) [reply .github/workflows/build.yml]: Ah, I see.

File: .github/workflows/build.yml (hunk 2 above)
Comment (tobim) [reply .github/workflows/build.yml]: Thanks for the explanation.

**Pull request merged**